waiter/src/waiter/metrics_sync.clj

```diff
@@ -116,13 +167,29 @@
                      (fn [existing-router-metrics]
                        (utils/deep-merge-maps (fn [x y] (or x y)) router-metrics
                                               (select-keys existing-router-metrics (keys router-metrics)))))
-          (assoc-in [:last-update-times source-router-id] time))
+          (assoc-in [:last-update-times source-router-id] time)
+
+          ; :external-metrics are merged based on 'updated-at' timestamp for individual instance metrics.
+          ; These metrics are absolute (one per waiter cluster, instead of waiter router), as they are provided
+          ; by an external source periodically with the /metrics/external endpoint.
+          (update :external-metrics merge-service-id->instance-id->metric-maps external-metrics query-state-fn))
       router-metrics-state)))
 
+(defn update-router-metrics-with-external-metrics
+  "Merges the service external metrics with existing external service metrics. External metrics are provided to the
+  waiter routers from another entity. These metrics must be merged based on the 'updated-at' timestamp."
+  [{:keys [router-id] :as router-metrics-state} incoming-service-id->instance-id->metric query-state-fn]
+  (with-catch
+    router-metrics-state
+    (-> router-metrics-state
+        (assoc-in [:last-update-times router-id] (du/date-to-str (t/now)))
```

Comment:
Should this be `router-id` or `external-metrics`? It feels strange to mix values for local metrics with that for external metrics.